r55/tests/evm-contract-call.rs

```diff
@@ -0,0 +1,195 @@
+    },
+};
+use revm::{
+    primitives::{address, Address},
+    InMemoryDB,
+};
+use tracing::{debug, error, info};
+
+const EVM_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/tests/simple-evm-contract.txt");
+const RISCV_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/../examples/evm-caller");
+
+// ------------------------------------------------------------------------------------------------
+//    SIMPLE EVM CONTRACT
+// ------------------------------------------------------------------------------------------------
+//    SPDX-License-Identifier: UNLICENSED
+//    pragma solidity ^0.8.13;
+//
+//    struct callParams { address target; bytes data; }
+//
+//    contract SimpleStorage {
+//        uint256 number;
+//
+//        function set(uint256 num) public { number = num; }
+//        function get() public view returns (uint256) { return number; }
+//        function rawCall(callParams calldata params) public returns (bool success) {
+//            (success, ) = params.target.call(params.data);
+//        }
+//    }
+// ------------------------------------------------------------------------------------------------
+
+#[test]
+fn evm_call() {
+    initialize_logger();
+
+    let mut db = InMemoryDB::default();
+
+    println!("PATH: {:#?}", EVM_PATH);
+    let bytecode_evm = load_bytecode_from_file(EVM_PATH);
+    let bytecode_r55 = compile_with_prefix(compile_deploy, RISCV_PATH).unwrap();
+    let evm = deploy_contract(&mut db, bytecode_evm, None).unwrap();
+    let r55 = deploy_contract(&mut db, bytecode_r55, None).unwrap();
+
+    let selector_get = get_selector_from_sig("get()");
+    let selector_set = get_selector_from_sig("set(uint256)");
+    let selector_raw_call = get_selector_from_sig("rawCall((address,bytes))");
+    let selector_x_get = get_selector_from_sig("x_get");
+    let selector_x_set = get_selector_from_sig("x_set");
+
+    let alice: Address = address!("000000000000000000000000000000000000000A");
+    add_balance_to_db(&mut db, alice, 1e18 as u64);
+
+    info!("----------------------------------------------------------");
+    info!("-- SET VALUE TX (EVM CONTRACT) ---------------------------");
+    info!("----------------------------------------------------------");
+    let value_set = U256::from(1e18);
+    let mut calldata_set = value_set.abi_encode();
+    let mut complete_calldata_set = selector_set.to_vec();
+    complete_calldata_set.append(&mut calldata_set);
+
+    debug!(
+        "Tx Calldata:\n> {:#?}",
+        Bytes::from(complete_calldata_set.clone())
+    );
+    match run_tx(&mut db, &evm, complete_calldata_set.clone()) {
+        Ok(res) => info!("{}", res),
+        Err(e) => {
+            error!("Error when executing tx! {:#?}", e);
+            panic!()
+        }
+    };
+
+    info!("----------------------------------------------------------");
+    info!("-- X-GET VALUE TX (R55 CONTRACT > EVM CONTRACT) ----------");
```

Comment:
```suggestion
    info!("-- X-GET VALUE TX (R55 CONTRACT -> EVM CONTRACT) ----------");
```

a find this a bit clearer, maybe can be changed in the instances below as well

Reply:
sure haha we can do that, i'll do a quick commit